### PR TITLE
#1403 make sure all pages are generated using html

### PR DESCRIPTION
**Pull request by krzyk**
#1403 
The bug was because browsers usually send `*/*` in their accept headers, and that was matching everyting in `FkTypes`

* corrected the accept header in `View.html`
* fixed that html should be shown only if the is no `application/vnd.zerocracy+xml` in the headers

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1407](https://github.com/zerocracy/farm/issues/1407) is now in scope, role is `REV`

**Comment by krzyk** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/zerocracy/farm/issues/1407#issuecomment-404155239))

@krzyk This is what I know about this job in [C3NDPUA8L](https://www.0crat.com/p/C3NDPUA8L), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#1407](https://github.com/zerocracy/farm/issues/1407) is [in scope](http://www.zerocracy.com/policy.html#14) for 3hr
  * The role is `REV`
  * The job is not assigned to anyone
  * The [budget](http://www.zerocracy.com/policy.html#4) is not set yet
  * These users are banned and won't be assigned:
    * @krzyk[/z](https://www.0crat.com/u/krzyk): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/C3NDPUA8L?q=%7Bjob%3A%27gh%3Azerocracy%2Ffarm%231407%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by 0crat** [comment]
This pull request [#1407](https://github.com/zerocracy/farm/issues/1407) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1921243933); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/zerocracy/tk/profile/TkAgendaTest.java [hunk 1]
```diff
@@ -97,9 +97,10 @@ public void rendersAgendaPage() throws Exception {
         agenda.add(project, job, "DEV");
         final String title = "Some random title is here";
         agenda.title(job, title);
+        final String html = new View(farm, "/u/yegor256/agenda").html();
         MatcherAssert.assertThat(
             XhtmlMatchers.xhtml(
-                new View(farm, "/u/yegor256/agenda").html()
+                html
```
Comment (amihaiemil) [inline src/test/java/com/zerocracy/tk/profile/TkAgendaTest.java]: @krzyk Why did you declare this variable? I don't see it used in more places :D and I think ``new View(farm, "/u/yegor256/agenda").html();`` is readable enough

File: src/main/java/com/zerocracy/tk/RsPage.java [hunk 2]
```diff
@@ -168,11 +172,26 @@ private static Response make(final Farm farm, final String xsl,
...
+                                    "application/vnd.zerocracy+xml"
+                                )
+                            ),
+                            request.head()
+                        )
+                    ).value(),
+                    () -> new Opt.Empty<>(),
+                    () -> new Opt.Single<>(html)
+                )
+            ).value(),
```
Comment (amihaiemil) [inline src/main/java/com/zerocracy/tk/RsPage.java]: @krzyk So if the header ``application/vnd.zerocracy+xml`` is present, then we return no header (``OptEmpty``), otherwise we return the response with ``text/html`` content type?

File: src/main/java/com/zerocracy/tk/RsPage.java [hunk 3]
```diff
@@ -168,11 +172,26 @@ private static Response make(final Farm farm, final String xsl,
...
             new FkTypes(
-                "application/vnd.zerocracy+xml",
+                "*/*",
```
Comment (amihaiemil) [inline src/main/java/com/zerocracy/tk/RsPage.java]: @krzyk Why this change here?

File: src/test/java/com/zerocracy/tk/View.java [hunk 4]
```diff
@@ -74,7 +74,7 @@ public String html() throws IOException {
         return this.page(
             // @checkstyle LineLength (1 line)
             "Mozilla/5.0 (X11; Linux x86_64; rv:62.0) Gecko/20100101 Firefox/62.0",
-            "text/html,application/xhtml+xml,application/xml"
+            "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8"
```
Comment (amihaiemil) [inline src/test/java/com/zerocracy/tk/View.java]: @krzyk See, this is I was asking to unit test this class :D apparently, it was not complete/correct.

File: src/main/java/com/zerocracy/tk/RsPage.java (hunk 2 above)
Comment (krzyk) [reply src/main/java/com/zerocracy/tk/RsPage.java]: @amihaiemil `RkFork` if it sees Empty, then it will try another fork.

File: src/main/java/com/zerocracy/tk/RsPage.java (hunk 3 above)
Comment (krzyk) [reply src/main/java/com/zerocracy/tk/RsPage.java]: @amihaiemil because it doesn't make sense to check for that header, this is a "catch-all" case (last fork) so we need to execute it either way.

File: src/main/java/com/zerocracy/tk/RsPage.java (hunk 2 above)
Comment (amihaiemil) [reply src/main/java/com/zerocracy/tk/RsPage.java]: @krzyk Could we test this code change somehow? I see now this ``request -> `` stuff is completely new :D

File: src/test/java/com/zerocracy/tk/View.java (hunk 4 above)
Comment (krzyk) [reply src/test/java/com/zerocracy/tk/View.java]: @amihaiemil yeah, but this wouldn't be caught in unit test :)
More like in some IT test using selenium, that would use real headers from a browser.

**Comment by krzyk** [comment]
@amihaiemil updated, please check

**Comment by amihaiemil** [comment]
@g4s8 Krzysztof said something interesting [here](https://github.com/zerocracy/farm/pull/1407#discussion_r201936944), maybe it would be nice to have Selenium tests? The app is not so big, so it wouldn't be that complicated. We'd only need a proper Test environment :)

**Comment by g4s8** [comment]
@amihaiemil @krzyk yes, it will be good to add Selenium tests, but we need to fix this bug first: https://github.com/zerocracy/farm/issues/655

**Comment by g4s8** [comment]
@rultor merge

**Comment by krzyk** [comment]
@g4s8 could you release new version, this agenda page in firefox is a real pain

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1407](https://github.com/zerocracy/farm/issues/1407) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by g4s8** [comment]
@krzyk sure, see https://github.com/zerocracy/farm/issues/1403#issuecomment-404440802

**Comment by ypshenychka** [comment]
@amihaiemil According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only two issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)
